src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -885,6 +896,11 @@ public static IResourceBuilder<T> WithUrlForEndpoint<T>(this IResourceBuilder<T>
             if (urlForEndpoint is not null)
             {
                 callback(urlForEndpoint);
+                if (urlForEndpoint.Endpoint is not null && urlForEndpoint.Url.StartsWith('/'))
+                {
+                    // Url is relative, update it to be absolute using the endpoint URL
+                    urlForEndpoint.Url = urlForEndpoint.Endpoint.Url.TrimEnd('/') + urlForEndpoint.Url;
+                }
```

Comment:
Right now this conversion of relative URLs to absolute URLs is only done in the `WithUrlForEndpoint` overloads. If a URL associated with an endpoint is added via the `WithUrls` callback, and the URL is relative, it will not be converted to absolute. I'm thinking I should probably change this logic to ensure all relative URLs associated with endpoints are converted to absolute URLs.